testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -1725,13 +1746,13 @@ void testCastToBoolean(CastType castType, SqlOperatorFixture f) {
     f.checkString("case 1 when 1 then cast('a' as varchar(1)) "
             + "when 2 then cast('bcd' as varchar(3)) end",
         "a", "VARCHAR(3)");
+    f.checkScalarExact("case 2 when 1 then 11.2 "
+            + "when 2 then 4.543 else null end",
+        "DECIMAL(5, 3)", "4.543");
     if (DECIMAL) {
-      f.checkScalarExact("case 2 when 1 then 11.2 "
-              + "when 2 then 4.543 else null end",
-          "DECIMAL(5, 3)", "4.543");
       f.checkScalarExact("case 1 when 1 then 11.2 "
               + "when 2 then 4.543 else null end",
-          "DECIMAL(5, 3)", "11.200");
+          "DECIMAL(5, 3)", "11.2");
```

Comment:
This test is still under DECIMAL flag. It is not fixed?

Reply:
Yes, this one still doesn't work. I will have to investigate why.
Note that here I am only implementing casts, so if there is a bug in the implementation of CASE it won't be affected.